lib/my_app.rb

```diff
@@ -0,0 +1,151 @@
+    current_user.save
+    redirect to('/new')
+  end
+
+  get '/new' do 
+    erb :newPost
+  end
+
+  get '/users/new' do
+    @user = User.new
+    erb :new_user
+  end
+
+  post '/users' do
+    @user = User.create(email: params[:email],
+                        password: params[:password],
+                        password_confirmation: params[:password_confirmation],
+                        user_name: params[:user_name])
+    if @user.save
+      session[:user_id] = @user.id
+      redirect to '/'
+    else
+      flash.now[:errors] = @user.errors.full_messages
+      erb :new_user
+    end
+  end
+
+  get '/sessions/new' do
+    erb :'sessions/new'
+  end
+
+  get '/booked/:id' do 
+    space = Space.last(id: params[:id])
+    if current_user != nil
+      owner = space.user.last.user_name
+      user = current_user.user_name
+      @booking_object = Booking.create(name: user, owner: owner, room: space.name, confirmation: params[:false])
+      session[:booking_ex] = @booking_object.id
+      @booking = Booking.last(id: session[:booking_ex])
+    end
+      erb :confirmation
+  end
+
+  get '/booking-requests' do 
+    if current_user
+      @bookings = Booking.all(owner: current_user.user_name)
+    end
+    erb :booking_requests
+  end
+
+  get '/book' do 
+    @bookings_placed = Booking.all
+    erb :'book'
+  end
+
+
+  post '/sessions' do
+    user = User.authenticate(params[:email], params[:password])
+    if user
+      session[:user_id] = user.id
+      redirect to '/'
+    else
+      flash.now[:errors] = ['The email or password is incorrect']
+      erb :'sessions/new' 
+    end
+  end
+
+  delete '/sessions' do
+    session[:user_id] = nil
+    session[:booking_ex] = nil
+    flash.keep[:notice] = 'goodbye!'
+    redirect to '/'
+  end
+
+  get '/users/listings' do
+    if current_user != nil
+      @place = current_user.space
+    end
+    erb :user_listings
+  end
+
+  get '/confirm/:id' do 
+    @booking = Booking.last(id: params[:id])
+    @booking.update(confirmation: 'Booking confirmed')
+    @booking.save
+    redirect to '/booking-requests'
+  end
+
+  get '/deny/:id' do 
+    @booking = Booking.last(id: params[:id])
+    @booking.update(confirmation: 'Booking denied')
+    @booking.save
+    redirect to '/booking-requests'
+  end
+
+
+  get '/edit/:id' do
+    @edit_space = Space.last(id: params[:id])
+    session[:get_id] = @edit_space.id
+    erb :'edit/edit_listings'
+  end
+
+  post '/edit' do
+    update_space = Space.last(id: session[:get_id])
+    update_space.update(params)
+    redirect to '/'
+  end
+
+  post '/delete' do
+    delete_space = Space.get(session[:get_id])
+    relationship = SpaceUser.first(space_id: session[:get_id], user_id: current_user.id)
+    relationship.destroy!
+    delete_space.destroy
+    redirect to '/'
+  end
+
+  # Do not put any controllers after this to avoid loop
+  get '/:id' do
+    @detail = Space.last(id: params[:id])
+    erb :details
+  end
```

Comment:
I don't think this is a good idea at all. This route should read `get '/spaces/:id'`, as `Space` is a RESTful resource.